### PR TITLE
Library: date exceptions

### PR DESCRIPTION
**Pull request by rerowep**
* NEW: add date exceptions in the library JSON schema
* NEW: update editor to manage library

Co-Authored-by: Bertrand Zuchuat <bertrand.zuchuat@rero.ch>
Co-Authored-by: Nicolas Labat <nicolas.labat@rero.ch
Signed-off-by: Peter Weber <Peter.Weber@rero.ch>

How to test:
Run the scripts `bootstrap` and `setup`

The library "Bibliothèque cantonale valdôtaine, site de Pont-Saint-Martin"
has predefined exception dates.  
* check detailed library view
* edit a library and add or change exception dates

### CODE REVIEW

File: ui/admin/package.json [hunk 1]
```diff
@@ -24,7 +24,8 @@
     "angular-in-memory-web-api": "^0.7.0",
     "bootstrap": "^4.1.1",
     "core-js": "^2.5.4",
-    "ngx-bootstrap": "^3.1.1",
+    "moment": "^2.22.2",
+    "ngx-bootstrap": "3.0.1",
```
[the remark this answers is not in the document]
Comment (rerowep) [reply ui/admin/package.json]: We need it for DatePicker

File: ui/admin/src/app/libraries/library.ts [hunk 2]
```diff
@@ -32,4 +48,36 @@ export class Library {
...
+    Object.assign(this, obj);
+  }
+
+  deleteException(index) {
+    this.exception_dates.splice(index, 1);
+  }
+
+  sortExceptions() {
+    this.exception_dates.sort(function(a, b) {
+        const keyA = new Date(a.start_date);
```
Comment (jma) [inline ui/admin/src/app/libraries/library.ts]: moment can be used. Something like: `return moment(a.start_date).substract(moment(b.start_date))`

File: rero_ils/modules/organisations_libraries/utils.py [hunk 3]
```diff
@@ -66,6 +66,16 @@ def save_library(data, record_type, record_class, parent_pid=None):
         library = record_class.get_record_by_pid(pid)
         library.update(data, dbcommit=False, reindex=False)
     else:
+        if 'opening_hours' not in data:
+            data['opening_hours'] = [
```
[the remark this answers is not in the document]
Comment (rerowep) [reply rero_ils/modules/organisations_libraries/utils.py]: Will be deleted with task #743

File: ui/admin/src/app/libraries/exception-dates-edit/exception-dates-edit.component.ts [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+  onCancel() {
+    this.bsModalRef.hide();
+    return false;
+  }
+
+  onPeriodChange(period) {
+    this.form.is_period.setValue(period);
+    if (period) {
+      for (let i = 0; i < this.form.times.length; i++) {
```
[the remark this answers is not in the document]
Comment (rerowep) [reply ui/admin/src/app/libraries/exception-dates-edit/exception-dates-edit.component.ts]: No, it means count of time in times

**Pull request merged**